services/wallet/reader.go

```diff
@@ -479,13 +479,13 @@ func (r *Reader) GetWalletToken(ctx context.Context, clients map[uint64]chain.Cl
 
 	var (
 		group             = async.NewAtomicGroup(ctx)
-		prices            = map[string]map[string]float64{}
+		prices            = map[string]map[string]market.DataPoint{}
 		tokenDetails      = map[string]thirdparty.TokenDetails{}
 		tokenMarketValues = map[string]thirdparty.TokenMarketValues{}
 	)
 
 	group.Add(func(parent context.Context) error {
-		prices, err = r.marketManager.FetchPrices(tokenSymbols, currencies)
+		prices, err = r.marketManager.GetOrFetchPrices(tokenSymbols, currencies, 60)
```

Comment:
Shall you update other references, actually i am not sure if FetchPrices should be ever used anymore?

Reply:
Hmm the only other interesting usage of `FetchPrices` seems to be here: https://github.com/status-im/status-go/blob/seanstrom/fix-get-wallet-token-rate-limits/services/wallet/router/sendtype/send_type.go#L52

Do you know if that should be updated?

Reply:
I'd say yes, @saledjenic to confirm.

Reply:
Correct, we need fresh prices in that case, so according to these changes, maybe it would be better to use `GetOrFetchPrices` with `maxAgeInSeconds = 0` instead (if I get it well, the used price may be up to 60 seconds old)?

Reply:
Actually it depends on how precise we want to be in displaying tx and bonder fees in fiat value, cause if we're good to use 60 seconds old price to display them we're good, otherwise, it would be good to have the latest price.

Reply:
@saledjenic Yeah good points, I'm not sure how fresh the data needs to be when preparing a transaction, but I imagine it would be nice to have something fairly fresh, so maybe 60 seconds would be too long.

For example, when I'm building a quote in Metamask, the fees and estimates are re-calculated every 30 seconds. So maybe we could say that the cache could only be 30 seconds old at the most when preparing a transaction?.

In comparison, when the app is displaying the wallet balance or token balances, we may not need very fresh data, so 60 seconds might be okay.

---

At the moment, 60 seconds is a guess for what's okay for the GetWalletToken function. But I think we can decide together what the ideal amount would be for:
* Displaying balances
* Preparing transactions
* other areas of the app that show overviews of assets
* other areas of the app that are about to perform an important action for assets

cc @ilmotta

Reply:
> So maybe we could say that the cache could only be 30 seconds old at the most when preparing a transaction?

We have the fees update mechanism in place, that updates fees for a chain based on the speed of the new block creation for that chain, so for the Mainnet chain our fees are being updated every ~12 seconds. That roughly means we update fees ~5 times in a minute and every time we're using the same price. I am fine with using up to 30 seconds prices for that purpose or maybe use the block interval creation time instead https://github.com/status-im/status-go/blob/develop/services/wallet/router/router_updates.go#L13-L15 ?